pkg/cmd/server/start/start_node.go

```diff
@@ -249,30 +273,39 @@ func (o NodeOptions) IsRunFromConfig() bool {
 	return (len(o.ConfigFile) > 0)
 }
 
-func StartNode(nodeConfig configapi.NodeConfig) error {
+func StartNode(nodeConfig configapi.NodeConfig, nodeArgs *NodeArgs) error {
```

Comment:
just pass in the `Components`... want to severely constrain what can override config